src/main/java/emissary/core/BaseDataObject.java

```diff
@@ -194,6 +194,10 @@ public class BaseDataObject implements Serializable, Cloneable, Remote, IBaseDat
 
     final SafeUsageChecker safeUsageChecker = new SafeUsageChecker();
 
+    protected final IBaseDataObject parent;
+
+    protected final Set<String> inheritedKeys;
```

Comment:
Recommend removing this field from the BaseDataObject` and passing as a parameter to `IBaseDataObjectHelper#getParametersWithInheritance()`